RobotsWantedLeague/Services/RobotsService.cs

```diff
@@ -1,59 +1,50 @@
+using Microsoft.Extensions.Options;
+using MongoDB.Driver;
+
 namespace RobotsWantedLeague.Services;
 
 using RobotsWantedLeague.Models;
 
 public class RobotsService: IRobotsService
 {
-    private readonly List<Robot> robots;
-    private int idGenerator = 0;
-    public List<Robot> Robots { get => robots; }
+    private readonly IMongoCollection<Robot> robotsCollection;
+
+    public List<Robot> Robots { get => robotsCollection.AsQueryable().ToList(); }
 
-    public RobotsService()
+    public RobotsService(IOptions<RobotsDatabaseSettings> bookStoreDatabaseSettings)
     {
-        robots = new List<Robot>();
-    }
+        var mongoClient = new MongoClient(
+            bookStoreDatabaseSettings.Value.ConnectionString);
+
+        var mongoDatabase = mongoClient.GetDatabase(
+            bookStoreDatabaseSettings.Value.DatabaseName);
 
-    private int generateId(){
-        idGenerator = idGenerator + 1;
-        return idGenerator;
+        this.robotsCollection = mongoDatabase.GetCollection<Robot>(bookStoreDatabaseSettings.Value.RobotsCollectionName);
     }
 
     public Robot CreateRobot(string name,
                           int weight,
                           int height,
                           string country)
     {
-        var robot = new Robot(generateId(), name, weight, height, country);
-        robots.Add(robot);
+        var robot = new Robot(null, name, weight, height, country);
+        robotsCollection.InsertOne(robot);
         return robot;
     }
+    
 
-    private int getIndexOfRobotById(int id){
-        int idx = 0;
-        foreach (Robot robot in robots){
-            if (robot.Id == id){
-                return idx;
-            }
-            idx ++;
-        }
-        return -1;
-    }
-
-    public Robot? GetRobotById(int id){
-        int indexToDelete = getIndexOfRobotById(id);
-        if(indexToDelete == -1){
-            return null;
-        }
-        return robots[indexToDelete];
+    public Robot? GetRobotById(string id)
+    {
+        IQueryable<Robot> q = from robot in robotsCollection.AsQueryable()
+            where robot.Id == id
+            select robot;
+        return q.First();
     }
 
-    public bool DeleteRobotById(int id){
-        int indexToDelete = getIndexOfRobotById(id);
-        if(indexToDelete == -1){
-            return false;
-        }
-        robots.RemoveAt(indexToDelete);
-        return true;
+    public bool DeleteRobotById(string id)
```

Comment:
Deleting is a little bit more complicated.  The weird part is the lambda function `robot => robot.Id == id`.